packages/node/e2e/e2e-test.spec.ts

```diff
@@ -1,246 +1,157 @@
-//         expect(response.status).toBe(401);
-//       });
-//     });
-
-//     describe("Post-logout fetch", () => {
-//       it("can fetch a public resource after logging out", async () => {
-//         const session = new Session();
-//         await session.login({
-//           clientId,
-//           clientSecret,
-//           oidcIssuer,
-//         });
-//         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-//         const publicResourceUrl = session.info.webId!;
-//         await session.logout();
-//         const response = await session.fetch(publicResourceUrl);
-//         expect(response.status).toBe(200);
-//         await expect(response.text()).resolves.toContain(
-//           ":PersonalProfileDocument"
-//         );
-//       });
-
-//       it("cannot fetch a private resource after logging out", async () => {
-//         const session = new Session();
-//         const privateResourceUrl = rootContainer;
-//         await session.login({
-//           clientId,
-//           clientSecret,
-//           oidcIssuer,
-//         });
-//         await session.logout();
-//         const response = await session.fetch(privateResourceUrl);
-//         expect(response.status).toBe(401);
-//       });
-//     });
-//   }
-// );
+      response = await unauthSession.fetch(privateResourceUrl);
+      expect(response.status).toBe(200);
+
+      await unauthSession.logout();
+    });
+  });
+
+  describe("Unauthenticated fetch", () => {
+    it("can fetch a public resource when not logged in", async () => {
+      // FIXME: This is only required to get a WebID. Once test setup is automated,
+      // this should be deleted.
+      const publicResourceUrl = authenticatedSession.info.webId as string;
+
+      const unauthenticatedSession = new Session();
+      const response = await unauthenticatedSession.fetch(publicResourceUrl, {
+        headers: {
+          Accept: "text/turtle",
+        },
+      });
+      expect(response.status).toBe(200);
+      await expect(response.text()).resolves.toContain(
+        ":PersonalProfileDocument"
+      );
+    });
+
+    it("cannot fetch a private resource when not logged in", async () => {
+      const unauthenticatedSession = new Session();
+      const privateResourceUrl = ENV.pod;
+      const response = await unauthenticatedSession.fetch(privateResourceUrl);
+      expect(response.status).toBe(401);
+    });
+  });
+
+  describe("Post-logout fetch", () => {
+    it("can fetch a public resource after logging out", async () => {
+      const publicResourceUrl = authenticatedSession.info.webId as string;
+      await authenticatedSession.logout();
+      const response = await authenticatedSession.fetch(publicResourceUrl, {
+        headers: {
+          Accept: "text/turtle",
+        },
+      });
+      expect(response.status).toBe(200);
+      await expect(response.text()).resolves.toContain(
+        ":PersonalProfileDocument"
+      );
+    });
```

Comment:
```suggestion
    it("can fetch the WebID after logging out", async () => {
      const publicResourceUrl = authenticatedSession.info.webId as string;
      await authenticatedSession.logout();
      const response = await authenticatedSession.fetch(publicResourceUrl, {
        headers: {
          Accept: "text/turtle",
        },
      });
      expect(response.status).toBe(200);
      await expect(response.text()).resolves.toContain(
        ":PersonalProfileDocument"
      );
    });
```

This doesn't really test "a public resource" it tests the WebID (not necessarily hosted on a Solid server).

Reply:
Right, I have a bunch of FIXMEs for this, I'll add one here.

Reply:
Done in 38864feb0ba578c8684e4caeccab899e09048b66. I want to have resource setup and teardown there, so it's definitely out of scope for this PR